akka-sample-fsm-java/src/main/java/sample/Hakker.java

```diff
@@ -0,0 +1,150 @@
+package sample;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import sample.Chopstick.ChopstickAnswer;
+import sample.Chopstick.ChopstickMessage;
+
+import java.time.Duration;
+
+/*
+ * A hakker is an awesome dude or dudette who either thinks about hacking or has to eat ;-)
+ */
+class Hakker {
+
+    interface HakkerMessage { }
+
+    enum Eat implements HakkerMessage {
+        INSTANCE
+    }
+
+    enum Think implements HakkerMessage {
+        INSTANCE
+    }
+
+    static class ChopstickAnswerAdaptor implements HakkerMessage {
+        final ChopstickAnswer msg;
+
+        public ChopstickAnswerAdaptor(ChopstickAnswer msg) {
+            this.msg = msg;
+        }
+    }
+    private final ActorContext<HakkerMessage> ctx;
+    private final String name;
+    private final ActorRef<ChopstickMessage> left;
+    private final ActorRef<ChopstickMessage> right;
+
+    public Hakker(ActorContext<HakkerMessage> ctx, String name, ActorRef<ChopstickMessage> left, ActorRef<ChopstickMessage> right) {
+        this.ctx = ctx;
+        this.name = name;
+        this.left = left;
+        this.right = right;
+    }
+
+    Behavior<HakkerMessage> waiting() {
+        return Behaviors.receive(HakkerMessage.class)
+                .onMessage(Think.class, (msg) -> {
+                    ctx.getLog().info("{} starts to think", name);
+                    return startThinking(Duration.ofSeconds(5));
+                })
+                .build();
+    }
+
+    //When a hakker is thinking it can become hungry
+    //and try to pick up its chopsticks and eat
+    Behavior<HakkerMessage> thinking() {
+        ActorRef<ChopstickAnswer> adapter = ctx.messageAdapter(ChopstickAnswer.class, ChopstickAnswerAdaptor::new);
```

Comment:
might be better to create the adapter once

Reply:
Doh, changed that for the scala but not the java